File: backend/src/main/java/cl/redvecinal/backend/user/service/UserService.java
package cl.redvecinal.backend.user.service;

import cl.redvecinal.backend.user.dto.request.UpdateProfileDto;
import cl.redvecinal.backend.user.dto.response.UserMyProfileDto;
import cl.redvecinal.backend.user.dto.response.UserProfileDto;

public interface UserService {
    boolean isUserRegistered(String phoneNumber);
    void resetPassword(String phoneNumber, String password);
    UserMyProfileDto showMyProfile();
    UserProfileDto showProfileUser(Long id);
    void updateMyProfile(UpdateProfileDto request);
}

File: frontend/lib/features/user/services/user_service.dart
import 'package:dio/dio.dart';
import 'package:frontend/core/network/api_client.dart';
import 'package:frontend/features/auth/models/response/phone_verification_response.dart';
import 'package:retrofit/retrofit.dart';
import 'package:frontend/features/user/models/request/update_profile_request.dart';
import 'package:frontend/features/user/models/response/my_profile_response.dart';
import 'package:frontend/features/user/models/response/other_profile_response.dart';
import 'package:frontend/features/auth/models/request/reset_password_request.dart';

part 'user_service.g.dart';

@RestApi(baseUrl: "/api/users")
abstract class UserService {
  factory UserService(Dio dio, {String baseUrl}) = _UserService;
  factory UserService.withDefaults() => UserService(ApiClient.instance);

  @GET("/consult-phoneNumber")
  Future<PhoneVerificationResponse> consultPhoneNumber(
    @Query('phoneNumber') String phoneNumber,
  );

  @PATCH("/reset-password")
  Future<void> resetPassword(
    @Body() ResetPasswordRequest resetPasswordRequest,
    @Header("Reset-Token") String resetToken,
  );

  @GET("/profile/me")
  Future<MyProfileResponse> getMyProfile();

  @GET("/profile/{id}")
  Future<OtherProfileResponse> getUserProfile(@Path("id") int id);

  @PATCH("/profile/me")
  Future<void> updateMyProfile(
    @Body() UpdateProfileRequest updateProfileRequest,
  );
}
